vispy/visuals/markers.py

```diff
@@ -47,9 +47,18 @@
 
     if (u_scaling == true) {
         // calculate point size from visual to framebuffer coords to determine size
-        vec4 x = $framebuffer_to_visual(fb_pos + vec4(big_float, 0, 0, 0));
-        x = (x - pos);
-        vec4 size_vec = $visual_to_framebuffer(pos + normalize(x) * a_size);
+        // move horizontally in framebuffer space
+        // then go to scene coordinates (not visual, so scaling is accounted for)
+        vec4 x = $framebuffer_to_scene(fb_pos + vec4(big_float, 0, 0, 0));
+        // subtract position, so we get the scene-coordinate vector describing
+        // an "horizontal direction parallel to the screen"
+        vec4 scene_pos = $framebuffer_to_scene(fb_pos);
+        x = (x - scene_pos);
+        // multiply that direction by the size (in scene space) and add it to the position
+        // this gives us the position of the edge of the point, which we convert in screen space
+        vec4 size_vec = $scene_to_framebuffer(scene_pos + normalize(x) * a_size);
```

Comment:
Doesn't this mean that size is now in screen coordinates and not visual coordinates? Or are we saying since it is a multiplier/factor it doesn't matter?

Reply:
No, size used to be in `visual` coordinates, but it's now in `scene` coordinates (not screen). This basically only changes that scaling is accounted. Note that the calculation happens in scene space (where "size" has a physical meaning), and not in framebuffer space.

Apart from the bugfix above, a consequence of this PR is that now the visual scaling does not affect the marker size; meaning a marker of size `10` will look exactly the same size if you set `scale = 20`, while the *coordinates* will be scaled... I think that's desirable, but it might affect some applications.

Note that you can't both fix this bug and prevent the above without somehow deforming markers into 3D ellipsoid (and worse with other marker symbols), which would basically require a new shader altogether.

Reply:
> Note that you can't both fix this bug and prevent the above

Yeah that's where I thought this was going.

Playing devil's advocate: if the Visual had a non-linear transform on it, wouldn't this mean that the size would not be affected by this transform? I'm not sure I care to try to handle this any better than what you've done, but I'm sure someone will file a bug for it in the next...10 years.

Reply:
Yeah, these are two different use cases that are probably incompatible... I guess I could actually add another flag that lets wou switch to the previous behaviour? Though that's also easy to add later if needed.

I had never realized this before, but this actually annoyed me more than once before in napari, because I used `scale` to rescale data to the correct pixel size, and had to change the marker size by a factor of `scale` to counteract it. So this is a double bugfix from my point of view :P

Reply:
I'm fine with it as-is. Feel free to merge.